Python/Product/PythonTools/PythonTools/Intellisense/CompletionAnalysis.cs

```diff
@@ -103,7 +113,7 @@ public class CompletionAnalysis {
 
         internal AnalysisEntry GetAnalysisEntry() {
             AnalysisEntry entry;
-            if (TextBuffer.TryGetAnalysisEntry(out entry) && entry != null) {
+            if (_view.TryGetAnalysisEntry(TextBuffer, _serviceProvider, out entry) && entry != null) {
```

Comment:
Not really, they're determined by some combination of the two.  In general we use the buffers filename grabbed from the ITextDocument property to find the analyzer.  If the file is in a project then the project analyzer will be used and that's that.  

Sometimes the view also influences it though, but we can't just rely upon a property on the buffer.  For the diff view case for example you end up w/ a left view, a right view, and an inline view.  We can use the diff factory to figure out that we're in a diff view, and then if one file is a project file and one is not, we can open both in the project context, with the temporary file that's outside of the project being marked as such so that we'll throw it away.

If I then went and opened that temporary file in a non-diff view we'd throw it into the default analyzer and we'd have 2 different buffer parsers working on it and 2 different analyzers operating on it.  

Likewise REPL window buffers are also directly influenced by their view - in that case we have buffers that have no file name except for the fake filename we create for them.  We treat them all as the same file, so all of the buffers get combined and analyzed as a single AST but parsed individually.  

Other projection buffers should work similarly but it may take some special handling just like it does for other weird views like the diff view.  But there's enough odd combinations where we need to consider both the view and the buffer for actually understanding how something gets analyzed.